Bottom: 20da1c7373ce22b26b293db83f3a460fb9dedc35
Top:    35d57b8123a5e38613688dda6e9fe4d28cb11133
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-07-05 17:50:39 +0100

Refresh of move-arc-from-included-angle

---

diff --git a/src/misc.c b/src/misc.c
index 83daec5..ec37c3f 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -319,6 +319,12 @@ SetLineBoundingBox (LineType *Line)
   SetPointBoundingBox (&Line->Point2);
 }
 
+/* ---------------------------------------------------------------------------
+ * calculates an arc center and radius for the arc described by
+ * two points, and an included angle between them. Return locations for
+ * cx, cy are required. radius, start_angle and delta_angle are optional
+ * and may be NULL if not required.
+ */
 void
 calc_arc_from_points_and_included_angle (PointType *p1, PointType *p2, Angle included_angle,
                                          Coord *cx, Coord *cy, Coord *radius,
@@ -371,7 +377,7 @@ calc_arc_from_points_and_included_angle (PointType *p1, PointType *p2, Angle inc
                     (p1->Y - *cy) * (p1->Y - *cy));
 
   if (start_angle != NULL)
-    *start_angle = TO_DEGREES (atan2 ((p1->Y - v->cy), -(p1->X - v->cx)));
+    *start_angle = TO_DEGREES (atan2 ((p1->Y - *cy), -(p1->X - *cx)));
 
   if (delta_angle != NULL)
     *delta_angle = included_angle;
